src/baconian.rs

```diff
@@ -0,0 +1,467 @@
+    ///
+    /// Where ...
+    ///
+    /// * The encoding will be distinct for all alphabetical characters, or classical
+    ///     where I, J, U and V are mapped to the same value pairs
+    /// * An optional decoy message that will will be used to hide the message -
+    ///     default is boilerplate "Lorem ipsum" text.
+    ///
+    fn new(key: (bool, Option<String>)) -> Result<Baconian, &'static str> {
+        Ok(Baconian {
+            distinct: key.0,
+            decoy_text: key.1.unwrap_or_else(|| String::from(DEFAULT_DECOY)),
+        })
+    }
+
+    /// Encrypt a message using the Baconian cipher
+    ///
+    /// * The message to be encrypted can only be ~18% of the decoy_text as each character
+    ///     of message is encoded by 5 encoding characters `AAAAA`, `AAAAB`, etc.
+    /// * The italicised ciphertext is then hidden in a decoy text, where, for each 'B'
+    ///     in the ciphertext, the character is italicised in the decoy_text.
+    ///
+    /// # Examples
+    /// Basic usage:
+    ///
+    /// ```
+    /// use cipher_crypt::{Cipher, Baconian};
+    ///
+    /// let b = Baconian::new((false, None)).unwrap();
+    /// let message = "Hello";
+    /// let cipher_text = "Lo𝘳𝘦𝘮 ip𝘴um d𝘰l𝘰r s𝘪t 𝘢me𝘵, 𝘯e 𝘵";
+    ///
+    /// assert_eq!(cipher_text, b.encrypt(message).unwrap());
+    /// ```
+    ///
+    fn encrypt(&self, message: &str) -> Result<String, &'static str> {
+        let mut non_alphas = 0; // A counter for non_alphas
+
+        for c in self.decoy_text.chars() {
+            if !c.is_alphabetic() {
+                non_alphas += 1;
+            }
+        }
+        // Check whether the message fits in the decoy
+        // Note: that non-alphabetical characters will be skipped.
+        if (message.len() * CODE_LEN) > self.decoy_text.len() - non_alphas {
+            return Err("Message too long for supplied decoy text.");
+        }
+
+        let mut secret = String::new();
+        // Iterate through the message encoding each char
+        // Ignore non-alphabetical chars
+        for c in message.chars() {
+            // get code and add to secret
+            let mut key = String::new();
+            key.push(c);
+            secret += &get_code(self.distinct, &key);
+        }
+
+        // Complex: decoy_slice needs to = secret.len + num_non_alphabetical_chars
+        let mut decoy_slice = self.decoy_text.clone();
+        let mut alphas = 0;
+        non_alphas = 0;
+        for c in self.decoy_text.chars() {
+            if c.is_alphabetic() {
+                alphas += 1;
+            } else {
+                non_alphas += 1;
+            }
+            if alphas == secret.len() {
+                break;
+            }
+        }
+        decoy_slice.truncate(alphas + non_alphas);
+
+        let mut decoy_msg = String::new();
+        for c in decoy_slice.chars() {
+            if c.is_alphabetic() {
+                match secret.remove(0) {
+                    'B' => {
```

Comment:
Some comments would be nice to explain what is happening here

Reply:
Done